### PR TITLE
JENKINS-22283 Show upstream

### PR DESCRIPTION
**Pull request by patbos**
Will show upstream jobs triggered the pipeline. In this example below the Package job is configured to as the first job in the pipeline and jobs CompA and CompB have triggered the Package.

screen shot 2016-06-16 at 13 14 21

### CODE REVIEW

File: src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java [hunk 1]
```diff
@@ -235,41 +236,66 @@ public Pipeline createPipelineAggregated(ItemGroup context, boolean showAggregat
...
+    public List<Pipeline> createPipelineLatest(int noOfPipelines, ItemGroup context, boolean pagingEnabled,
+                                               boolean showUpstream) throws PipelineException {
         List<Pipeline> result = new ArrayList<Pipeline>();
-        int no = noOfPipelines;
-        if (firstProject.isInQueue()) {
-            String pipeLineTimestamp = PipelineUtils.formatTimestamp(firstProject.getQueueItem().getInQueueSince());
-            List<Stage> pipelineStages = new ArrayList<Stage>();
-            for (Stage stage : getStages()) {
-                pipelineStages.add(stage.createLatestStage(context, null));
+        if (showUpstream) {
```
Comment (tommysdk) [inline src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java]: Could we refactor this block and extract to its own method? Same goes for the else block.

**Comment by patbos** [comment]
@tommysdk Done

File: src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Will show upstream jobs for the first job.
```
Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: how then job could be first if it not first?

Comment (tommysdk) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @KostyaSha the first job in the pipeline could have upstream dependencies that could trigger a new build of that job. See example screenshot above.

Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @tommysdk @patbos could you check email?

Comment (tommysdk) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @KostyaSha have we missed something?

Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @tommysdk i had few critical questions in terms of time when i searched for answers.

Comment (tommysdk) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @KostyaSha ok. Must have gotten lost somewhere. Hope you got your answers already?

Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: No.

Comment (tommysdk) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @KostyaSha could you try again please?

**Comment by tommysdk** [comment]
@patbos should we proceed with this?

File: src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html (hunk 2 above)
Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: CC'd into email.

Comment (KostyaSha) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @tommysdk did you get it?

Comment (tommysdk) [inline src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/RegExpSpec/help-showUpstream.html]: @KostyaSha yes! Got it know!

**Comment by patbos** [comment]
Will close this PR since the merge is too complicated, have done a new branch from master and will reimplement the feature.

**Comment by tommysdk** [comment]
@patbos Got it!

**Pull request closed without merging**